src/main/java/spoon/support/sniper/internal/MutableTokenWriter.java

```diff
@@ -22,8 +22,38 @@
 	private final TokenWriter delegate;
 	private boolean muted = false;
 
+	// indentation style to use for new elements
+	private boolean originSourceUsesTabulations;
+	private int originSourceTabulationSize;
+
 	public MutableTokenWriter(Environment env) {
-		this.delegate = new DefaultTokenWriter(new PrinterHelper(env));
+		this.delegate = new DefaultTokenWriter(new SniperPrinterHelper(env));
+		originSourceUsesTabulations = true;
+		originSourceTabulationSize = 1;
+	}
+
+	private class SniperPrinterHelper extends PrinterHelper {
+		private final Environment env;
+
+		SniperPrinterHelper(Environment env) {
+			super(env);
+			this.env = env;
+		}
+
+		/**
+		 * We override this method to use the correct style of indentation for new elements.
+		 */
+		@Override
+		protected void autoWriteTabs() {
+			int setTabulationSize = env.getTabulationSize();
+			env.useTabulations(originSourceUsesTabulations);
+			env.setTabulationSize(originSourceTabulationSize);
+
+			super.autoWriteTabs();
+
+			env.setTabulationSize(setTabulationSize);
+			env.useTabulations(true);
+		}
```

Comment:
This is the actual thing that overrides the printing of tabs with whatever indentation style is present in the source file. I'm not completely satisfied with this solution as it seems a bit hacky, but I couldn't figure out a better way to do it.